src/pyedb/dotnet/clr_module.py

```diff
@@ -34,79 +33,62 @@ def custom_show_warning(message, category, filename, lineno, file=None, line=Non
-            try:
-                dotnet_root = find_dotnet_root()
-                runtime_config = find_runtime_config(dotnet_root)
-            except Exception:
-                warnings.warn(
-                    "Unable to set DOTNET root and locate the runtime configuration file. "
-                    "Falling back to using dotnetcore2."
-                )
-                warnings.warn(LINUX_WARNING)
-
-                import dotnetcore2
+    dotnet_root = None
+    runtime_config = None
+    # Use system .NET core runtime or fall back to dotnetcore2
+    if os.environ.get("DOTNET_ROOT") is None:
+        try:
+            from clr_loader import get_coreclr
 
-                dotnet_root = Path(dotnetcore2.__file__).parent / "bin"
-                runtime_config = pyedb_path / "misc" / "pyedb.runtimeconfig.json"
+            runtime = get_coreclr()
+            load(runtime)
+            os.environ["DOTNET_ROOT"] = str(runtime.dotnet_root)
+            is_clr = True
+        # TODO: Fall backing to dotnetcore2 should be removed in a near future.
+        except Exception:
+            warnings.warn(
+                "Unable to set .NET root and locate the runtime configuration file. "
+                "Falling back to using dotnetcore2."
+            )
+            warnings.warn(LINUX_WARNING)
+
+            import dotnetcore2
+
+            dotnet_root = Path(dotnetcore2.__file__).parent / "bin"
+            runtime_config = pyedb_path / "misc" / "pyedb.runtimeconfig.json"
+    # Use specified .NET root folder
+    else:
+        dotnet_root = Path(os.environ["DOTNET_ROOT"])
+        # Patch the case where DOTNET_ROOT leads to dotnetcore2 for more information
+        # see https://github.com/ansys/pyedb/issues/922
+        # TODO: Remove once dotnetcore2 is deprecated
+        if dotnet_root.parent.name == "dotnetcore2":
+            runtime_config = pyedb_path / "misc" / "pyedb.runtimeconfig.json"
         else:
-            dotnet_root = Path(os.environ["DOTNET_ROOT"])
-            try:
-                runtime_config = find_runtime_config(dotnet_root)
-            except Exception as e:
+            from clr_loader import find_runtimes
+
+            candidates = [rt for rt in find_runtimes() if rt.name == "Microsoft.NETCore.App"]
+            candidates.sort(key=lambda spec: spec.version, reverse=True)
+            if not candidates:
                 raise RuntimeError(
                     "Configuration file could not be found from DOTNET_ROOT. "
                     "Please ensure that .NET SDK is correctly installed or "
                     "that DOTNET_ROOT is correctly set."
                 )
+            runtime_config = candidates[0]
+    # Use specific .NET core runtime
+    if dotnet_root is not None and runtime_config is not None:
         try:
             load("coreclr", runtime_config=str(runtime_config), dotnet_root=str(dotnet_root))
+            os.environ["DOTNET_ROOT"] = dotnet_root
```

Comment:
This line will fail if we have entered the except block at line 51 to fall back to dotnetcore2 as dotnet_root is set to a Path object on line 60 and we do not convert to str here (though we should probably use `Path.as_posix()` instead).  I believe this is occurring on the toolkit test runners if we do not initialize PyAEDT first (thus ensuring that DOTNET_ROOT is already set).  I can provide a link to a run demonstrating this.